pkg/didcomm/transport/http/inbound.go

```diff
@@ -88,3 +89,49 @@ func validateHTTPMethod(w http.ResponseWriter, r *http.Request) bool {
 
 	return true
 }
+
+// Inbound http type.
+type Inbound struct {
+	server *http.Server
+}
+
+// NewInbound creates a new HTTP inbound transport instance.
+func NewInbound(addr string) (*Inbound, error) {
+	if addr == "" {
+		return nil, errors.New("http address is mandatory")
+	}
+
+	return &Inbound{server: &http.Server{Addr: addr}}, nil
+}
+
+// Start the http server.
+func (i *Inbound) Start(prov transport.InboundProvider) error {
+	handler, err := NewInboundHandler(prov)
+	if err != nil {
+		return fmt.Errorf("HTTP server start failed: %w", err)
+	}
+
+	i.server.Handler = handler
+
+	go func() {
+		if err := i.server.ListenAndServe(); err != http.ErrServerClosed {
+			logger.Fatalf("HTTP server start with address [%s] failed, cause:  %s", i.server.Addr, err)
+		}
+	}()
+
+	return nil
+}
+
+// Stop the http server.
+func (i *Inbound) Stop() error {
+	if err := i.server.Shutdown(context.TODO()); err != nil {
```

Comment:
why context.TODO instead of context.Background?

(I assume you are planning to make this cancellable?)

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/6f6d39856a36112ab2f3116510c390f4c2855da9..1183264d25a8e484f015ea047b61c78c7a982cd5?diff=split#diff-18bf71ef9d572ae715312cab65be25eeR127)